lib/util/traverser.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @fileoverview Wrapper around estraverse
+ * @author Nicholas C. Zakas
+ * @copyright 2016 Nicholas C. Zakas. All rights reserved.
+ * See LICENSE in root directory for full license.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var estraverse = require("estraverse");
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+var KEY_BLACKLIST = [
+    "parent",
+    "leadingComments",
+    "trailingComments"
+];
+
+/**
+ * Wrapper around an estraverse controller that ensures the correct keys
+ * are visited.
+ * @constructor
+ */
+function Traverser() {
+
+    var controller = Object.create(new estraverse.Controller()),
+        originalTraverse = controller.traverse;
+
+    // intercept call to traverse() and add the fallback key to the visitor
+    controller.traverse = function(node, visitor) {
+        visitor.fallback = Traverser.getKeys;
+        return originalTraverse.call(this, node, visitor);
```

Comment:
I had meant, why not have `Traverser` itself have a prototype equal to `Object.create(estraverse.Controller.prototype)` and override. But you are right about needing to have the extra call stack layer. For some reason I was thinking the visitor keys logic could be set on the prototype rather than passed to the `controller.traverse` function. Thanks for the clarification.